features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/featureflags/internal/model/DatadogEvaluationContextTest.kt

```diff
@@ -0,0 +1,316 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.featureflags.internal.model
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.flags.featureflags.model.EvaluationContext
+import com.datadog.android.flags.utils.forge.ForgeConfigurator
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class DatadogEvaluationContextTest {
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Test
+    fun `M create normalized context W from() { valid attributes only }`(forge: Forge) {
+        // Given
+        val targetingKey = forge.anAlphabeticalString()
+        val stringValue = forge.anAlphabeticalString()
+        val numberValue = forge.anInt()
+        val booleanValue = forge.aBool()
+
+        val publicContext = EvaluationContext(
+            targetingKey = targetingKey,
+            attributes = mapOf(
+                "stringAttr" to stringValue,
+                "numberAttr" to numberValue,
+                "booleanAttr" to booleanValue
+            )
+        )
+
+        // When
+        val datadogContext = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        assertThat(datadogContext).isNotNull
```

Comment:
minor trick: `checkNotNull(datadogContext)` can be used here (it also throws assertion error) and then you can benefit from smart case to non-nullable type below.

Same can be done for the tests below.